test/utilities/translator.js

```diff
@@ -16,67 +16,139 @@
+                        throw new Error('Options preset not found');
+                    }
+                    const preset2 = optionsPresets[entry];
+                    if (preset2.type !== type) {
+                        throw new Error('Invalid options preset type');
+                    }
+                    Object.assign(preset, structuredClone(preset2));
                 }
-                Object.assign(options, structuredClone(optionsPresets[entry]));
                 break;
             case 'object':
-                Object.assign(options, structuredClone(entry));
+                if (entry.type !== type) {
+                    throw new Error('Invalid options preset type');
+                }
+                Object.assign(preset, structuredClone(entry));
                 break;
             default:
                 throw new Error('Invalid options type');
         }
     }
+    return preset;
+}
 
-    // Construct regex
-    if (Array.isArray(options.textReplacements)) {
-        options.textReplacements = options.textReplacements.map((value) => {
-            if (Array.isArray(value)) {
-                value = value.map(({pattern, flags, replacement}) => ({pattern: new RegExp(pattern, flags), replacement}));
-            }
-            return value;
-        });
+
+/**
+ * @param {string} dictionaryName
+ * @param {import('test/translator').OptionsPresetObject} optionsPresets
+ * @param {import('test/translator').OptionsList} optionsArray
+ * @returns {import('translation').FindKanjiOptions}
+ */
+export function createFindKanjiOptions(dictionaryName, optionsPresets, optionsArray) {
+    const preset = getCompositePreset('kanji', optionsPresets, optionsArray);
+
+    /** @type {import('translation').KanjiEnabledDictionaryMap} */
+    const enabledDictionaryMap = new Map();
+    const presetEnabledDictionaryMap = preset.enabledDictionaryMap;
+    if (Array.isArray(presetEnabledDictionaryMap)) {
+        for (const [key, value] of presetEnabledDictionaryMap) {
+            enabledDictionaryMap.set(key === placeholder ? dictionaryName : key, value);
+        }
     }
 
-    // Update structure
-    const placeholder = '${title}';
-    if (options.mainDictionary === placeholder) {
-        options.mainDictionary = dictionaryName;
+    return {
+        enabledDictionaryMap,
+        removeNonJapaneseCharacters: !!preset.removeNonJapaneseCharacters
+    };
+}
+
+/**
+ * @param {string} dictionaryName
+ * @param {import('test/translator').OptionsPresetObject} optionsPresets
+ * @param {import('test/translator').OptionsList} optionsArray
+ * @returns {import('translation').FindTermsOptions}
+ */
+export function createFindTermsOptions(dictionaryName, optionsPresets, optionsArray) {
+    const preset = getCompositePreset('terms', optionsPresets, optionsArray);
+
+    /** @type {import('translation').TermEnabledDictionaryMap} */
+    const enabledDictionaryMap = new Map();
```

Comment:
I guess you decided that it wasn't worth reusing/breaking out these few lines of identical logic?

Reply:
I tried getting it to work with types, and it realistically should, but Typescript wasn't cooperating for whatever reason. It was treating the templates as unions instead of a single discrete type, which I don't understand why it was doing that.

I can add that code back in, but it involves some ugly casting, which is why I eventually gave up on it. But I don't feel strongly about that decision either way.